src/Dom/Browser.Dom.fs

```diff
@@ -485,6 +485,41 @@ type [<AllowNullLiteral>] Element =
     abstract scroll : ScrollToOptions -> unit
     abstract scrollBy : ScrollToOptions -> unit
     abstract scrollTo : ScrollToOptions -> unit
+    /// Returns the open shadow root that is hosted by the element, or null if no open shadow root is present.
+    abstract shadowRoot: ShadowRoot with get
+    /// Attaches a shadow DOM tree to the specified element and returns a reference to its ShadowRoot.
+    abstract attachShadow: ShadowRootInit -> ShadowRoot
+
+type [<AllowNullLiteral>] ShadowRootInit =
+    /// A string specifying the encapsulation mode for the shadow DOM tree. This can be one of: 'open' or 'closed'
+    /// a closed shadow root will deny access to the node(s) from JavaScript outside
+    abstract mode: string with get, set
+    /// A boolean that, when set to true, specifies behavior that mitigates custom element issues around focusability.
+    /// When a non-focusable part of the shadow DOM is clicked, the first focusable part is given focus,
+    /// and the shadow host is given any available :focus styling.
+    abstract delegatesFocus: bool with get, set
+
+/// The ShadowRoot interface of the Shadow DOM API is the root node of a DOM subtree
+/// that is rendered separately from a document's main DOM tree.
+/// You can retrieve a reference to an element's shadow root using its Element.shadowRoot property,
+/// provided it was created using Element.attachShadow() with the mode option set to open.
+type [<AllowNullLiteral>] ShadowRoot =
+    inherit DocumentFragment
+    inherit Element
```

Comment:
This is strange, the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot) includes more properties and methods for ShadowRoot than the |whatwg specs](https://dom.spec.whatwg.org/#shadowroot), not sure which one should we follow. In any case, both sites say `ShadowRoot` inherits `DocumentFragment` but not `Element`.

BTW, for readonly properties, it's ok to omit `with get`.

Reply:
Yeah I didn't find anything that was reliable in that case,also for reference there is the typings.d.ts

https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L4725
https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L14024
https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L9468

I'm not entirely sure how inheritance is done with typings.d.ts but inspecting manually in the editor gives me methods from these

- Node
- ParentNode
- DocumentOrShadowRoot
- EventTarget
- DocumentFragment
- InnerHTML

should I make it inherit from all of the above rather than just Element?

I chose Element because using it it's just like using any other element and I think internally the browser is giving you a reference to the element that hosts the shadowRoot